### PR TITLE
refactor(material/button): handle disabled states through tokens

### PR DESCRIPTION
**Pull request by crisbeto**
MDC applies disabled styles through the `:disabled` selector which forced us to work around it by using a selector like `.mat-mdc-button[disabled][disabled]` and setting both the disabled and enabled button tokens. This is problematic, because it increases the specificity too much and it introduces space for mistakes, because we have to duplicate the tokens values.

These changes resolve the issue by re-applying the token slots to disabled buttons with the correct selector so that they can reuse the same tokens as the `button` nodes.

### CODE REVIEW

File: src/material/button/_button-base.scss [hunk 1]
```diff
@@ -89,15 +88,3 @@
       $query: mdc-helpers.$mdc-base-styles-query);
   }
 }
-
-// Changes a button token set to exclude the ripple styles.
-@function mat-private-button-remove-ripple($tokens) {
```
Comment (crisbeto) [inline src/material/button/_button-base.scss:94]: Unrelated change, I just noticed that we weren't using this.

**Review by andrewseguin — APPROVED** [review]

**Comment by crisbeto** [comment]
Merging this manually through the UI. Will send a follow-up to cherry-pick into the RC branch.
